### PR TITLE
Setting up initial GitHub actions, to test for Ruff formatting violations

### PR DESCRIPTION
**Pull request by tmarshall**
# Why

This will serve as the foundation of other tests, including various version tests.

### CODE REVIEW

**Comment by tmarshall** [comment]
hard to tell if this will work 100% when it lands. tried to use some 3rd party tools to test, but they were not working properly, so 🤞🏻

File: pyproject.toml [hunk 1]
```diff
@@ -23,6 +23,7 @@ python = "^3.10"
 anomalo = "^0.0.7"
 
 [tool.poetry.dev-dependencies]
+poetry = "^1.7.0"
```
Comment (kfoxb) [inline pyproject.toml]: Is this needed? Is this to install/upgrade to a specific version of poetry?

File: .github/workflows/python.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  check-python-code-quality:
+    runs-on: ubuntu-latest
+    timeout-minutes: 45
+
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v4
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
```
Comment (kfoxb) [inline .github/workflows/python.yml:23]: You can use poetry with this action and it'll handle caching dependencies automatically for you. See [here](https://github.com/actions/setup-python/blob/main/docs/advanced-usage.md#caching-packages). Then you won't need `poetry install` below.

File: .github/workflows/python.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        with:
+          python-version: 3.11
+
+      - name: Install dependencies
+        run: poetry install
+
+      - name: Install ruff
+        run: pip install ruff
```
Comment (kfoxb) [inline .github/workflows/python.yml]: Can this be added to the pyproject.toml so we don't have to have a one off install?

File: pyproject.toml (hunk 1 above)
Comment (tmarshall) [reply pyproject.toml]: good point, don't think it's needed

File: .github/workflows/python.yml (hunk 2 above)
Comment (tmarshall) [reply .github/workflows/python.yml:23]: example still has `poetry install`:

```
steps:
- uses: actions/checkout@v4
- name: Install poetry
  run: pipx install poetry
- uses: actions/setup-python@v4
  with:
    python-version: '3.9'
    cache: 'poetry'
- run: poetry install
- run: poetry run pytest
```

**Review by kfoxb — APPROVED** [review]

**Pull request merged**